server/net_listener_unix.go

```diff
@@ -0,0 +1,35 @@
+//go:build !windows
+
+package server
+
+import (
+	"context"
+	"net"
+	"syscall"
+
+	"golang.org/x/sys/unix"
+)
+
+func newNetListener(protocol, address string) (net.Listener, error) {
+	lc := net.ListenConfig{
+		Control: func(network, address string, c syscall.RawConn) error {
+			var socketErr error
+			err := c.Control(func(fd uintptr) {
+				err := unix.SetsockoptInt(int(fd), unix.SOL_SOCKET, unix.SO_REUSEADDR, 1)
```

Comment:
(minor) Would be nice to leave a short one sentence comment in here quickly explaining the intent of this. Would be helpful for future code readers who find this and aren't sure why it's needed. (More helpful to explain the issue we were seeing with port reuse and the hypothesis about TIME_WAIT than just rehashing what the options do, since that's easy to look up.)

Reply:
Fixed.